### ISSUE TITLE
CLI responses to user errors are long and unhelpful

### ISSUE THREAD

**Issue by allanday** [flatpak/flatpak#1290]
If a user gets a command wrong, the CLI response is usually to spit out the `--help` text followed by a short error message. The `--help` text is often really long. It comes as a surprise, is hard to parse, makes it hard to find the error message, and isn't always useful.

A better approach would be to show the error message and then suggest using `--help`, instead of showing its output. For example:

```
$ flatpak remote-indo flathub org.gimp.GIMP
'remote-indo' is not a flatpak command. See 'flatpak --help' for information.
```

This way you're not overwhelming the user with too much information.

An even better approach would be to actually make the error messages helpful. Like:

```
$ flatpak remote-indo flathub org.gimp.GIMP
'remote-indo' is not a flatpak command. Did you mean 'remote-info'?

See 'flatpak --help' for information.
```

**Comment by mwleeds** [comment]
Here's another example of that behavior:
```
$ git committ
git: 'committ' is not a git command. See 'git --help'.

Did you mean this?
	commit
```

I like the way you've written it better, but if there are two commands with equally similar names there's not a great way to choose which to print.

**Comment by allanday** [comment]
> I like the way you've written it better, but if there are two commands with equally similar names there's not a great way to choose which to print.

That's a good point and you could use a list like git does. git's phrasing is a little staccato though, so maybe you could do it like:

```
$ flatpak remote-indo flathub org.gimp.GIMP
'remote-indo' is not a flatpak command (see 'flatpak --help').

Suggested commands:
        remote-info
```

**Comment by matthiasclasen** [comment]
This was done
